src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -62,6 +67,22 @@ export class DatasetService {
     return this.defaultDataset;
   }
 
+  public getRecentDataset(datasetID: string | undefined): Dataset | undefined {
+    if (datasetID) {
+      return this.recentDatasets.get(datasetID);
+    }
+  }
+
+  public getRecentDatasets(): Dataset[] {
+    return Array.from(this.recentDatasets.values());
+  }
+
+  public addRecentDataset(dataset: Dataset | undefined): void {
```

Comment:
do you think instead of relying on the UI component for setting the dataset the recent dataset is the one that is used? so like nobody could add recent dataset they can only access the recent data sets?

Reply:
+1 Again not a blocker for this PR, but we should add a fast follow to do this automatically when a dataset is set in the query manager. additionally if we dont want certain setQuery.dataset calls to update the LRU cache, we can always pass a option to not cache it. This can be used for on page load updates that load the default dataset on page load if no dataset is present in the URL. 

This will be a source of bugs in the future when someone in another app or usecase forgets to call this.